### ISSUE TITLE
Flexvolume plugin error

### ISSUE THREAD

**Issue by zimme** [rancher/k3os#282]
I'm seeing this in my k3s logs.
```
W1116 12:10:13.254151    1893 probe.go:268] Flexvolume plugin directory at /usr/libexec/kubernetes/kubelet-plugins/volume/exec/ does not exist. Recreating.
E1116 12:10:13.254450    1893 plugins.go:600] Error initializing dynamic plugin prober: Error (re-)creating driver directory: mkdir /usr/libexec/kubernetes: read-only file system
```
I'm guessing it's because `/usr`is the squashfs mounted as `ro`. Should `/usr/libexec` be writeable like `/usr/lib/firmware` is?

**Comment by dweomer** [comment]
@zimme yes, `/usr` is sourced from the squashfs embedded in the `/k3os/system/k3os/current/k3os` executable.

hmm, based off of https://github.com/kubernetes/community/blob/master/contributors/devel/sig-storage/flexvolume.md#prerequisites you should be able to override with `--flex-volume-plugin-dir` (not sure how you're deploying, helm?)

that said, i wonder if we could just be more friendly by setting up a symlink to somewhere under `/var/lib`

**Comment by dweomer** [comment]
> @zimme yes, `/usr` is sourced from the squashfs embedded in the `/k3os/system/k3os/current/k3os` executable.
> 
> hmm, based off of https://github.com/kubernetes/community/blob/master/contributors/devel/sig-storage/flexvolume.md#prerequisites you should be able to override with `--flex-volume-plugin-dir` (not sure how you're deploying, helm?)
> 
> that said, i wonder if we could just be more friendly by setting up a symlink to somewhere under `/var/lib`

good candidate opportunity for leverage a `boot_cmd` :smile:

**Comment by zimme** [comment]
This is on a clean k3os install, I haven't deployed flexvolume myself

**Comment by dweomer** [comment]
this `config.yaml` did the trick:
```
k3os:
  k3s_args:
  - "--kube-controller-arg=flex-volume-plugin-dir=/var/lib/kubernetes/kubelet-plugins/volume/exec"
```

**Comment by zimme** [comment]
I'm thinking that k3os should provide a default config for this or synlink the default path to somethink writable. As this comes with k3s/k3os by default

**Comment by dweomer** [comment]
> I'm thinking that k3os should provide a default config for this or synlink the default path to something writable. As this comes with k3s/k3os by default

yes, `/usr/libexec/kubernetes` could be easily pointed to `/var/lib/kubernetes` but because it is so simple and doesn't seem necessary i am inclined to let he user/operator deal with it. i'll ask @ibuildthecloud what he thinks after kubecon

**Comment by zimme** [comment]
@dweomer, did @ibuildthecloud have an opinion on this?

**Comment by dweomer** [comment]
I’ve been meaning to just default it via a symlink in /usr space

**Comment by dweomer** [comment]
Fixed via #447
